### DISCUSSION TITLE
🌐 Help translate Note Toolbar

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#118] [General]
### I'd love your help translating Note Toolbar into other languages, or at least reviewing them!

I'm currently looking for:

1. Anybody who would like to help translate Note Toolbar's UI to another language (that's supported by Obsidian), or review an existing translation.
2. If anybody has had experience contributing translations before, I'd love to hear how that went, and your process.

[Instructions here](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/118#discussioncomment-10548222)

Down the road I'll be looking into opening up the User Guide to allow for translations as well.

Thank you for your continued support for Note Toolbar!

**Comment by chrisgurney** [comment]
@moyf Absolutely no pressure, but if you're interested in contributing translations for Note Toolbar, or know anybody else who might, I'd be grateful!

**Reply by Moyf** [reply]
No problem, I think I can assist you with the Chinese translation for this plugin  ; )

**Reply by chrisgurney** [reply]
@Moyf Thank you so much! I'll let you know when I have a first version of the English file done, then you can take your time.

**Reply by chrisgurney** [reply]
FYI @Moyf no pressure but if you ever have the time or want to translate the `styles.css` for Style Settings, below is how @laktiv did it for Ukrainian. Again, absolutely no pressure; I'm sure somebody else might be interested!

https://github.com/chrisgurney/obsidian-note-toolbar/commit/565cdfe99c9aaafd10414c5e7993cb40a255f2f4#diff-380b7b38760dd442e897eb0164c58f6a17da966ccaca6318017a468c163979b1

**Reply by Moyf** [reply]
> FYI @Moyf no pressure but if you ever have the time or want to translate the `styles.css` for Style Settings, below is how @laktiv did it for Ukrainian. Again, absolutely no pressure; I'm sure somebody else might be interested!
> 
> [565cdfe#diff-380b7b38760dd442e897eb0164c58f6a17da966ccaca6318017a468c163979b1](https://github.com/chrisgurney/obsidian-note-toolbar/commit/565cdfe99c9aaafd10414c5e7993cb40a255f2f4#diff-380b7b38760dd442e897eb0164c58f6a17da966ccaca6318017a468c163979b1)

This time there is quite a lot of content, and I really don't know how I would quickly complete this translation work without AI, haha.  
Sorry to keep you waiting, here it is! #136

**Comment by chrisgurney** [comment]
### [Help translate Note Toolbar 🌐](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Help-translate-Note-Toolbar-%F0%9F%8C%90)

**Reply by chrisgurney** [reply]
@Moyf This should be ready to get started, whenever you have time of course! Absolutely no rush!

Should you have any questions, feel free to drop a note here or message me on the Obsidian Discord `@cheznine`

Thanks again!

**Reply by chrisgurney** [reply]
Chinese (Traditional) is now available in [1.12-beta-6](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.12-beta-6).

**Comment by laktiv** [comment]
Hi man.
Thank you for the beautiful plugin. And many thanks for the documentation. 
I've added Ukrainian translation. 
Also I have translated CSS file for Style plugin, but I have lack of skills to finish the transformation. See attached file with text.
Have a great day
[note_toolbar_css.json](https://github.com/user-attachments/files/16904331/note_toolbar_css.json)

**Reply by chrisgurney** [reply]
@laktiv Thank you so much for doing this! I have to say, it feels like an honor having the plugin translated into another language.

I'll get the UI strings incorporated right away.

As for the translations for CSS / Style Settings, thank you so much for those as well -- I'll have to add that to my instructions! I may have hit a couple of possible character encoding issues incorporating those. If you can do a check when I get the commit in, that would be fantastic.

**Reply by chrisgurney** [reply]
@laktiv There's a bit of English here you missed in the Style Settings. There's a couple other spots, when you have the time.

Feel free to comment right on the lines of the commit, if that's easier:
https://github.com/chrisgurney/obsidian-note-toolbar/commit/565cdfe99c9aaafd10414c5e7993cb40a255f2f4#diff-380b7b38760dd442e897eb0164c58f6a17da966ccaca6318017a468c163979b1R333

![CleanShot 2024-09-06 at 15 36 17@2x](https://github.com/user-attachments/assets/3731471b-d41d-4254-846a-a2a154a33a54)

**Reply by laktiv** [reply]
Just uploaded updated translation. Unfortunately, I'm not a programmer and some actions are hard to run from the first try. Hope to see the beta update to continue the detailed read-out and fix.

Will update the CSS as well.

Thank you.

**Reply by laktiv** [reply]
[styles.zip](https://github.com/user-attachments/files/16946273/styles.zip)
Updated CSS. Works good

**Reply by chrisgurney** [reply]
@laktiv Thank you. I've updated the translations in the [latest beta release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.12-beta-13).

Going forward, please get the latest code (do another fork, or do a `git pull`) before updating any translations. I've made a couple changes, including moving the Style Settings code out of the CSS and into a separate YAML file.

**Reply by laktiv** [reply]
Hi Chris. Could you please link me to manual (mb "for dummies") what I need to do properly? Git is something hard for me, still can't get it. 
Thank you in advance!!

**Reply by laktiv** [reply]
Wanted to do it from the start. Please check attached zip.
You could write menu in separate file and transorm to css file without asking everybody to edit styles.css
[scss menu.zip](https://github.com/user-attachments/files/16987163/scss.menu.zip)

**Reply by chrisgurney** [reply]
Thanks for this @laktiv !

I've moved the style settings comment out of the CSS and into a separate file: `src/style-settings.yaml`:
https://github.com/chrisgurney/obsidian-note-toolbar/blob/master/src/style-settings.yaml

I reviewed your changes manually and incorporated differences in whitespace and quotes, so thanks for that!

**Reply by chrisgurney** [reply]
@laktiv I will be uploading all the translation files to [Crowdin](https://crowdin.com/) and _hopefully_ if I can get that working it has tools like spell checking and suggested translations that will help, and eliminate the need to update code.

**Reply by laktiv** [reply]
Hi Chris,
Really don't know how to do it properly. But here updated uk.json file which I tried to upload last week but still can't see at update.
Thank you for help to add the file.
[uk.json](https://github.com/user-attachments/files/17135635/uk.json)

**Comment by hassanzouhar** [comment]
Loving this plugin, so when you get ready to accept new translations feel free to ping me. Happy to contribute with a Norwegian translation.

**Reply by chrisgurney** [reply]
@hassanzouhar Thank you! Please feel free to contribute one now. I never got around to adding support for CrowdIn, so see if you can follow these instructions, and let me know if you have any questions:

https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Help-translate-Note-Toolbar-%F0%9F%8C%90

Feel free to just work on the UI string translations for your first submission.

Thanks again!

**Reply by hassanzouhar** [reply]
It's on, like donkey kong!🦍
